apps/bbsync/cc.py

```diff
@@ -0,0 +1,398 @@
+            collections = collections.union(stream.collections)
+
+        return list(collections)
+
+    def collection_component(self) -> Tuple[Optional[str], str]:
+        """
+        parse RHSCL PS component into collection and
+        Bugzilla component and return them as a tuple
+        """
+        matched_collections = [
+            collection
+            for collection in self.collections()
+            if self.ps_component.startswith(collection + "-")
+            or self.ps_component == collection
+        ]
+
+        if not matched_collections:
+            return None, self.ps_component
+
+        if len(matched_collections) > 1:
+            raise ProductDataError(
+                f"Components' {self.ps_component} prefix matches more than "
+                f"one valid collection: {', '.join(matched_collections)}"
+            )
+
+        collection = matched_collections[0]
+        if collection and self.ps_component != collection:
+            return collection, self.ps_component[len(collection) + 1 :]
+
+        return collection, self.ps_component
+
+    def ps2bz_component(self) -> str:
+        """
+        translate PS component to Bugzilla one
+        RH software collections special case
+        """
+        try:
+            collection, bz_component = self.collection_component()
+        except ProductDataError:
+            # ignore exception when component matches more than one RHSCL
+            # collection and just use the previous component instead
+            return self.ps_component
+
+        # additionally store collection
+        # which may be potentially None
+        self.collection = collection
+
+        return bz_component
+
+    def component_cc(self) -> List[str]:
+        """
+        generate CCs based on component
+        RH software collections special case
+        """
+        cc_list = self.ps_module_obj.component_cc.get(self.bz_component, [])
+        cc_list.extend(
+            self.ps_module_obj.component_cc.get(self.collection, [])
+            if getattr(self, "collection", None)
+            else []
+        )
+
+        return cc_list
+
+
+class CCBuilder:
+    """
+    Bugzilla flaw CC list array builder
+    """
+
+    def __init__(self, flaw: Flaw, old_flaw: Optional[Flaw] = None) -> None:
+        """
+        init stuff
+        parametr old_flaw is optional as there is no old flaw on creation
+        and if not set we consider the query to be a create query
+        """
+        self.flaw = flaw
+        self.old_flaw = old_flaw
+
+        # unless the flaw is being created
+        # we extract the previous data
+        if not old_flaw:
+            return
+
+        self.old_cc = json.loads(self.old_flaw.meta_attr.get("cc", "[]"))
+
+        srtnotes = json.loads(self.old_flaw.meta_attr.get("original_srtnotes", "{}"))
+        old_affects = [
+            (affect["ps_module"], affect["ps_component"])
+            for affect in srtnotes.get("affects", [])
```

Comment:
Very good question! This is maybe a bit of artifact of my SFM2 thinking which is probably outdated here.

There are actually two cases we call this. One is when you save a flaw and that means that there are no changes to the affects (it is a different API endpoint) and old ones equal the new ones. The other is when you save the affect which means that there may be a single affect added, removed, or modified.

There should be currently no option to do some bulk change of flaw and affect(s) at once while this was what I was first afraid of and therefore rather took the stored original affects. If it was, we need to save one before the other to the DB at least when creating or deleting the affects because the relations need to be created and that requires UUIDs. My bad that I was making it save against the bulk change which is however not possible. I will have a look and rework it.

Reply:
If you want you can keep it as it is in case we ever do have some sort of builk operations as you mentioned, I don't mind the approach just pointing out that for other people it might not be obvious, so I would at least document the why in a comment, but up to you if you prefer to use `old_flaw` instead